include/topgg/webhook.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include <dpp/dpp.h>
+#include <nlohmann/json.hpp>
+#include <functional>
+#include <string>
+
+namespace topgg {
+
+/**
+ * @brief Data received from a bot vote webhook
+ */
+struct bot_vote_data {
+    dpp::snowflake user_id;          ///< ID of the user who voted
+    std::string type;                ///< Type of vote (usually "upvote")
+    bool is_weekend;                 ///< Whether the vote was done during weekend
+    std::string query;               ///< Query string data if any
+    
+    static bot_vote_data from_json(const nlohmann::json& j) {
+        bot_vote_data data;
+        data.user_id = dpp::snowflake(j["user"].get<std::string>());
+        data.type = j["type"].get<std::string>();
+        data.is_weekend = j["isWeekend"].get<bool>();
+        if (j.contains("query")) {
+            data.query = j["query"].get<std::string>();
+        }
+        return data;
+    }
+};
+
+/**
+ * @brief Webhook manager for handling Top.gg vote notifications
+ */
+class webhook_manager {
+public:
+    /**
+     * @brief Construct a new webhook manager
+     * @param cluster Reference to the DPP cluster
+     * @param port Port to listen on
+     * @param auth_token Authentication token for webhook verification
+     */
+    webhook_manager(dpp::cluster& cluster, uint16_t port, const std::string& auth_token) 
+        : cluster_(cluster), port_(port), auth_token_(auth_token) {
+        
+        // Set up interaction handler for webhooks
+        cluster_.on_interaction_create([this](const dpp::interaction_create_t& event) {
```

Comment:
This function body should be moved to `webhook.cpp` as it does not depend on any generics.